network/p2p/gossip/bloom.go

```diff
@@ -65,11 +65,15 @@ func ResetBloomFilterIfNeeded(
 	bloomFilter *BloomFilter,
 	falsePositiveProbability float64,
 ) (bool, error) {
-	if bloomFilter.bloom.FalsePositiveProbability() < falsePositiveProbability {
+	numSeeds, numBytes := bloomFilter.bloom.Parameters()
+	// TODO: Precalculate maxEntries, as it is independent of the current state
+	// of the bloom filter.
+	maxEntries := bloom.EstimateEntries(numSeeds, numBytes, falsePositiveProbability)
```

Comment:
Shouldn't we be able to just calculate this once in `NewBloomFilter`?

Reply:
The `falsePositiveProbability` passed in here can be (and is in practice) different than the `falsePositiveProbability` passed in to `NewBloomFilter`.

Reply:
Addressing: https://github.com/ava-labs/avalanchego/pull/2591